src/fields/struct_field.rs

```diff
@@ -127,19 +127,7 @@ impl StructField {
                     _ => panic!("Unsupported array type."),
                 }
 
-                match &array.len {
-                    syn::Expr::Lit(expr_lit) => match &expr_lit.lit {
-                        syn::Lit::Int(lit_int) => {
-                            let array_len: usize = lit_int.base10_parse().unwrap();
-                            self.generate_array_deserializer(array_type_token, array_len)
-                        }
-                        _ => panic!("Unsupported array len literal."),
-                    },
-                    syn::Expr::Path(expr_path) => {
-                        self.generate_array_deserializer(array_type_token, &expr_path.path)
-                    }
-                    _ => panic!("Unsupported array len expression."),
-                }
+                self.generate_array_deserializer(array_type_token, &array.len.clone().into_token_stream())
```

Comment:
If we change `generate_array_deserializer` to have signature `fn generate_array_deserializer(&self, array_type_token: syn::TypePath, array_len: &syn::Expr)` then we can simple do 
```suggestion
                self.generate_array_deserializer(array_type_token, &array.len)
```
here :)